src/__tests__/type.js

```diff
@@ -495,3 +495,16 @@ test('ignored {backspace} in controlled input', async () => {
     keyup: 4 (52)
   `)
 })
+
+test('typing in an input with validated type works', async () => {
```

Comment:
Consider adding a case for multiple input types, like here https://github.com/testing-library/user-event/blob/master/src/__tests__/clear.js#L77